src/rpc/connection.rs

```diff
@@ -587,7 +587,9 @@ impl Connection {
 		// Get a database reference
 		let kvs = DB.get().unwrap();
 		// Specify the query parameters
-		let var = Some(self.vars.to_owned());
+		let mut new_vars = self.vars.to_owned();
+		new_vars.insert(key.0.clone(), val.clone());
+		let var = Some(new_vars);
```

Comment:
Why this change @rushmorem?

Reply:
To make `kvs.compute()` also check if the key we are trying to set is protected and return an error if it is [just like this fix to the local engines](https://github.com/surrealdb/surrealdb/pull/3216/files#diff-72ebdd785075ef7d626a4ca4486726b82d1459743bdc41a8c416b521c6139551R770). Otherwise `compute` only computes the value and sets it without checking if the key is actually protected. It makes sure we don't store protected keys.

Reply:
```suggestion
		let var = Some(map! {
			key.0.clone() => Value::None,
			=> &self.vars
		});
```
1. Could we keep this consistent with the other functions?
2. We don't need to clone the `val` as that particular value is computed and then stored for future use. We are only storing it on the `var` object to check that it doesn't conflict with a protected variable name. Hence the `Value::None`. Does that make sense?

Reply:
Absolutely! Thanks! Let me update the SDK too...

Reply:
Done @tobiemh.